### PR TITLE
refactor tests and some kraken

### PR DESCRIPTION
**Pull request by dpark01**
Begin refactor of some metagenomics bits and shuffle some unrelated long running unit tests to integration. Preparatory work on #308, which is not yet addressed, but might as well merge this much into master for now.

### CODE REVIEW

File: test/unit/test_tools_kraken.py [hunk 1]
```diff
@@ -14,32 +14,71 @@ class TestToolKraken(TestCaseWithTmp):
...
+        patcher = mock.patch('tools.CondaPackage', autospec=True)
         self.addCleanup(patcher.stop)
-        self.mock_jelly = patcher.start()
-        self.mock_jelly().install_and_get_path.return_value = tempfile.tempdir
+        self.mock_conda = patcher.start()
+        self.mock_conda.return_value.verify_install.return_value = "mock"
+        self.mock_conda.return_value.is_attempted.return_value = True
+        self.mock_conda.return_value.is_installed.return_value = True
+        self.mock_conda.return_value.require_executability = False
+        self.mock_conda.return_value.executable_path.return_value = "/dev/null"
```
Comment (yesimon) [inline test/unit/test_tools_kraken.py]: We should open a new issue for a generalized mock conda object.

File: test/integration/test_kraken.py [hunk 2]
```diff
@@ -28,22 +28,6 @@ def fastq_to_sam():
...
-    return [os.path.join(data_dir, f)
-            for f in ['zaire_ebola.1.fastq', 'zaire_ebola.2.fastq']]
-
-
-def input_bam_paths():
-    data_dir = join(util.file.get_test_input_path(), 'TestMetagenomicsViralMix')
-    return join(data_dir, 'test-reads.bam')
-
-
-@pytest.fixture(scope='session')
```
Comment (yesimon) [inline test/integration/test_kraken.py]: Why removal of `scope=session` here?

**Comment by yesimon** [comment]
Looks good

**Comment by dpark01** [comment]
Thanks for the proofread!

**Pull request merged**